cmd/types.go

```diff
@@ -0,0 +1,176 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/artifacts"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/pyxis"
+	log "github.com/sirupsen/logrus"
+)
+
+// resultWriter defines methods associated with writing check results.
+type resultWriter interface {
+	OpenFile(name string) (io.WriteCloser, error)
+	io.WriteCloser
+}
+
+// resultSubmitter defines methods associated with submitting results to Red HAt.
+type resultSubmitter interface {
+	Submit(context.Context) error
+}
+
+// pyxisClient defines pyxis API interactions that are relevant to check executions in cmd.
+type pyxisClient interface {
+	FindImagesByDigest(ctx context.Context, digests []string) ([]pyxis.CertImage, error)
+	GetProject(context.Context) (*pyxis.CertProject, error)
+	SubmitResults(context.Context, *pyxis.CertificationInput) (*pyxis.CertificationResults, error)
+}
+
+// newPyxisClient initializes a pyxisClient with relevant information from cfg.
+// If the the CertificationProjectID, PyxisAPIToken, or PyxisHost are empty, then nil is returned.
+// Callers should treat a nil pyxis client as an indicator that pyxis calls should not be made.
+func newPyxisClient(ctx context.Context, cfg certification.Config) pyxisClient {
```

Comment:
Perhaps this should be 'newAuthenticatedPyxisClient'? There is a valid use case for getting the pyxis client without authentication.

Reply:
Since it is package local, should be fine. Do as you wish.